core/api/src/main/java/org/trellisldp/api/BinaryService.java

```diff
@@ -72,15 +73,17 @@ default void setContent(IRI identifier, InputStream stream) {
      * @param identifier the binary object identifier
      * @param stream the content
      * @param metadata any user metadata
+     * @return the new completion stage
      */
-    void setContent(IRI identifier, InputStream stream, Map<String, String> metadata);
+    CompletableFuture<Void> setContent(IRI identifier, InputStream stream, Map<String, String> metadata);
 
     /**
      * Purge the content from its corresponding datastore.
      *
      * @param identifier the binary object identifier
+     * @return the new completion stage
      */
-    void purgeContent(IRI identifier);
+    CompletableFuture<Void> purgeContent(IRI identifier);
```

Comment:
This method is not used anywhere in the codebase. The idea was that, at some point, there would be a mechanism to complete purge this data from the server, either through a new HTTP endpoint or via some CLI utility. For now, I think it would be best to remove the method, and once we're ready to add the aforementioned utility, the method can be re-added.